d4tools/src/stat/main.rs

```diff
@@ -227,6 +227,51 @@ fn hist_stat(matches: ArgMatches) -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+fn parse_stat_thresholds(stat: &str) -> Result<Vec<u32>, Box<dyn std::error::Error>> {
+    let thresholds_str = &stat["perc_cov=".len()..];
+
+    let mut thresholds: Vec<u32> = thresholds_str
+        .split(',')
+        .map(|s| s.trim().parse())
+        .collect::<Result<Vec<u32>, _>>()?;
+    thresholds.sort_unstable();
+    Ok(thresholds)
+}
+
+fn perc_cov_stat(
+    matches: ArgMatches,
+    mut print_header: bool,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let stat = matches.value_of("stat").unwrap_or("perc_cov=10,20,30");
+
+    let thresholds = parse_stat_thresholds(stat)?;
+    let mut unused = Vec::new();
+    let results = open_file_parse_region_and_then(matches, &mut unused, |mut input, regions| {
+        let tasks: Vec<_> = regions
+            .into_iter()
+            .map(|(chr, begin, end)| PercentCov::new(&chr, begin, end, thresholds.clone()))
+            .collect();
+
+        Ok(PercentCov::create_task(&mut input[0], tasks)?.run())
+    })?;
+
+    if print_header {
+        print!("#Chr\tStart\tEnd");
+        for t in thresholds.iter() {
+            print!("\t{}x", t);
+        }
+        println!();
+        print_header = false;
+    }
+    for r in results.into_iter() {
+        print!("{}\t{}\t{}", r.chrom, r.begin, r.end);
+        for value in r.output.iter() {
+            print!("\t{:.3}", value);
```

Comment:
I'm fine with this, but just noting that sometimes people will do huge ranges then it will say, for example `0.000` when there are bases above a given threshold. Might we use `:3g` or something?

Reply:
Makes sense to me, not familiar with `:3g` though and couldn't find anything about that. Could only find this on precision: https://doc.rust-lang.org/std/fmt/#precision

Reply:
I agree with the thought here, that information might be lost by truncating at three digits.

Seems the `g` formatting is present in the `printf` in C and bash, and automatically selects between precision and scientific notation https://www.gnu.org/software/libc/manual/html_node/Floating_002dPoint-Conversions.html. On a quick test it didn't seem to be supported in Rust though (I might be wrong).

What about using a sane default (for instance as you have now) and letting the user control the number of decimals / showing the full value through a flag?

(I am also fine with it as is though - eager to get this in quickly so we can get things going 😅 )

Reply:
Maybe something like this would work:

```
if value.abs() >= 0.001 {
    print!("\t{:.3}", value);
} else {
    print!("\t{:.3e}", value);
}
```

I.e. explicitly showing scientific notation for smaller numbers.

Reply:
Good idea. Sorry for delay got caught up with teaching things as the quarter starts soon. Will get this squared away in the morning!

Reply:
No worries, many thanks 🙏

Reply:
resolved in latest commits